src/west/app/project.py

```diff
@@ -1503,12 +1518,55 @@ def ensure_cloned(self, project, stats, take_stats):
             if take_stats:
                 stats['init'] = perf_counter() - start
 
+    def create_auto_cache_info(self, project, cache_dir):
+        # auto-cache helper: Create some short informal file with basic info
+        # about the local cache folder.
+        content = textwrap.dedent(f"""
+        The following local cache directory was automatically created by west:
+        - Local Cache:  {basename(cache_dir)}
+        - Project Url:  {project.url}
+        """)
+        with open(cache_dir + '.info', 'w') as f:
+            f.write(content)
+
+    def handle_auto_cache(self, project):
+        # update() helper. Initialize the specified cache directory if it has
+        # not been cloned yet. If the cache directory is already existing, it
+        # will be synced with remote.
+        auto_cache_dir = self.project_auto_cache(project)
+
+        if not auto_cache_dir:
+            # nothing to do: auto-cache directory was not specified
+            return
+
+        cache_dir = self.project_cache(project)
+        if cache_dir != auto_cache_dir:
+            # another cache was specified with higher priority.
+            return
+
+        if not Path(cache_dir).exists():
+            # The cache has not been created yet at the expected location.
+            # Ensure that its parent directory exist before cloning anything.
+            # Then clone the repository into the local cache.
+            cache_dir_parent = Path(cache_dir).parent
+            cache_dir_parent.mkdir(parents=True, exist_ok=True)
+            project.git(['clone', '--', project.url, os.fspath(cache_dir)],
+                        cwd=cache_dir_parent)
```

Comment:
If the project specified that it has submodules, we could already clone recursive maybe? Not sure if that would conflict with the `fetch` logic in the `else` case..